dartagnan/src/main/java/com/dat3m/dartagnan/program/memory/MemoryObject.java

```diff
@@ -34,17 +34,12 @@ public class MemoryObject extends IntExpr {
 
     private final HashMap<Integer, Expression> initialValues = new HashMap<>();
 
-    MemoryObject(int index, int size, boolean isStaticallyAllocated) {
-        super(TypeFactory.getInstance().getArchType());
+    MemoryObject(int index, IntegerType addressType, int size, boolean isStaticallyAllocated) {
+        super(addressType);
         this.index = index;
         this.size = size;
         this.isStatic = isStaticallyAllocated;
         this.isThreadLocal = false;
-
-        if (isStaticallyAllocated) {
-            // Static allocations are default-initialized
-            initialValues.put(0, ExpressionFactory.getInstance().makeZero(TypeFactory.getInstance().getArchType()));
-        }
```

Comment:
I was surprised that this caused no errors in the tests.  It should influence the result of `getStaticallyInitializedFields()` and thus the set of `Init` events of non-litmus programs.  Can this safely be removed?

Reply:
What problem do you expect? Static allocations are default initialized to zero in `C`, so why would this affect non-litmus programs? I guess the llvm parser already makes sure to do correct initialization so this code is likely redundant (i.e., nothing will change if you remove it).